### PR TITLE
Entity Data Model Changes to enable PATCH Functionality

### PR DESCRIPTION
**Pull request by eedwards0**
DO NOT merge with current PSC Filing Service Release 6 - PSC-163

### CODE REVIEW

**Review by PhilipGlover — COMMENTED** [review]
- I've run and submitted all requests for individual and corporate-entity
- all tests pass
- code looks good
- happy to approve once Rel 6 done

File: src/main/java/uk/gov/companieshouse/pscfiling/api/model/entity/PscIndividualFiling.java [hunk 1]
```diff
@@ -11,110 +13,67 @@
...
+                && Objects.equals(getStatementType(), that.getStatementType());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), getId(), getCountryOfResidence(), getDateOfBirth(),
+                getNameElements(), getNationality(), getResidentialAddress(),
+                getResidentialAddressSameAsCorrespondenceAddress(), getStatementActionDate(),
+                getStatementType());
+    }
```
Comment (hepsimo) [inline src/main/java/uk/gov/companieshouse/pscfiling/api/model/entity/PscIndividualFiling.java:51]: Please move these to after the getter methods to be more consistent with the other classes

File: src/main/java/uk/gov/companieshouse/pscfiling/api/model/entity/PscIndividualFiling.java [hunk 2]
```diff
@@ -11,110 +13,67 @@
...
     private Boolean residentialAddressSameAsCorrespondenceAddress;
     private LocalDate statementActionDate;
     private String statementType;
 
     public PscIndividualFiling() {
         // required by Spring JPA
-        pscCommon = PscCommon.builder()
+        PscCommon.builder()
                 .build();
     }
```
Comment (hepsimo) [inline src/main/java/uk/gov/companieshouse/pscfiling/api/model/entity/PscIndividualFiling.java:71]: Please remove this redundant builder call, but keep the public constructor and the comment

File: src/main/java/uk/gov/companieshouse/pscfiling/api/model/entity/PscIndividualFiling.java [hunk 3]
```diff
@@ -147,41 +106,24 @@ public String getStatementType() {
...
+                .add("etag='" + getEtag() + "'")
+                .add("kind='" + getKind() + "'")
+                .add("links=" + getLinks())
+                .add("naturesOfControl=" + getNaturesOfControl())
+                .add("notifiedOn=" + getNotifiedOn())
+                .add("referenceEtag='" + getReferenceEtag() + "'")
+                .add("referencePscId='" + getReferencePscId() + "'")
+                .add("registerEntryDate=" + getRegisterEntryDate())
+                .add("updatedAt=" + getUpdatedAt())
                 .add("countryOfResidence='" + countryOfResidence + "'")
```
Comment (hepsimo) [inline src/main/java/uk/gov/companieshouse/pscfiling/api/model/entity/PscIndividualFiling.java:127]: These lines can be replaced with
`.add(super.toString())`

**Review by hepsimo — CHANGES_REQUESTED** [review]
Changes look good. Some very minor fixes recommended.

**Review by hepsimo — APPROVED** [review]
Changes look good

**Pull request merged**